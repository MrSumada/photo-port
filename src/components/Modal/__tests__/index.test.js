import React from "react";
import { render, cleanup, fireEvent } from "@testing-library/react";
import '@testing-library/jest-dom/extend-expect';
import Modal from '..';

afterEach(cleanup);

const mockToggleModal= jest.fn();
const currentPhoto = {
    name: "Mr. Peanut",
    category: "portraits",
    description: "He sings, he dances, and he loves.",
    index: 1
}

describe('Modal component', () => {
    it('renders', () => {
        render(<Modal
            currentPhoto={currentPhoto}
            mockToggleModal={mockToggleModal}
        />);
    })

    it('matches snapshot', () => {
        const { asFragment } = render(<Modal 
            currentPhoto={currentPhoto}
            mockToggleModal={mockToggleModal}
        />);
        expect(asFragment()).toMatchSnapshot();
    })
})

describe('Click Event', () => {
    it('calls onClose handler', () => {
        const { getByText } = render(<Modal
            onClose={mockToggleModal}
            currentPhoto={currentPhoto}
        />);

        fireEvent.click(getByText('Close this modal'));

        expect(mockToggleModal).toHaveBeenCalledTimes(1);
    });
})