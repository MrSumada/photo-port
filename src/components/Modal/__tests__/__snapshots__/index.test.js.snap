// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="modalBackdrop"
  >
    <div
      class="modalContainer"
    >
      <h3
        class="modalTitle"
      >
        Mr. Peanut
      </h3>
      <img
        alt="current category"
        src="1.jpg"
      />
      <p>
        He sings, he dances, and he loves.
      </p>
      <button
        type="button"
      >
        Close this modal
      </button>
    </div>
  </div>
</DocumentFragment>
`;
